src/compiler/checker.ts

```diff
@@ -5993,6 +5993,7 @@ export function createTypeChecker(host: TypeCheckerHost): TypeChecker {
          * It also calls `setOriginalNode` to setup a `.original` pointer, since you basically *always* want these in the node builder.
          */
         function setTextRange<T extends Node>(context: NodeBuilderContext, range: T, location: Node | undefined): T {
+            if (range === location) return range;

```

Comment:
Move this below the copy check below - that copy needs to happen even if the range and location are the same node, originally.